src/engraving/dom/tuplet.cpp

```diff
@@ -839,4 +841,24 @@ int Tuplet::computeTupletDenominator(int numerator, Fraction totalDuration)
     }
     return ratio.numerator();
 }
+
+EngravingItem* Tuplet::nextElement()
+{
+    ChordRest* firstElement = toChordRest(elements().front());
+    if (firstElement->type() == ElementType::CHORD) {
+        Chord* chord = toChord(firstElement);
+        return chord->firstGraceOrNote();
+    } else {
+        return toRest(firstElement);
+    }
+    return firstElement;
+}
+
+EngravingItem* Tuplet::prevElement()
+{
+    ChordRest* firstElement = toChordRest(elements().front());
+    staff_idx_t staffId = firstElement->staffIdx();
+    EngravingItem* prevItem = firstElement->segment()->prevElement(staffId);
+    return prevItem;
+}
```

Comment:
Mu3 has a very different code here (but the same for nextElement except for that firstGraceOrNote();):
```cpp
Element* Tuplet::prevElement()
{
    ChordRest* lastElement = toChordRest(elements().back());
    if (lastElement) {
        if (lastElement->isChord())
            return toChord(lastElement)->notes().back();
        return toRest(lastElement);
    }
    return lastElement;
}
```